[CI] CoreWithSecurityClientYamlTestSuiteIT test {yaml=update/100_synthetic_source/stored text} failing

**Build Scans:**
- [elasticsearch-periodic #6258 / release-tests](https://gradle-enterprise.elastic.co/s/6ts4lc2j45ky2)
- [elasticsearch-pull-request #55698 / checkpart4_release-tests](https://gradle-enterprise.elastic.co/s/27ssy542foxn2)
- [elasticsearch-pull-request #55672 / checkpart4_release-tests](https://gradle-enterprise.elastic.co/s/rdvef6aipnmyw)
- [elasticsearch-pull-request #55646 / checkpart4_release-tests](https://gradle-enterprise.elastic.co/s/w7vnwjv7o5lp4)
- [elasticsearch-pull-request #55486 / checkpart4_release-tests](https://gradle-enterprise.elastic.co/s/ivmfy4pac6gpm)
- [elasticsearch-pull-request #55436 / part-4-entitlements](https://gradle-enterprise.elastic.co/s/uroommvfhmglm)
- [elasticsearch-pull-request #55424 / part-4-entitlements](https://gradle-enterprise.elastic.co/s/kxzwjrj3gk7fa)
- [elasticsearch-pull-request #55073 / part-4](https://gradle-enterprise.elastic.co/s/5hpg7jcmvzlsi)
- [elasticsearch-pull-request #54605 / checkpart4_release-tests](https://gradle-enterprise.elastic.co/s/msp5ooslwhz5e)

**Reproduction Line:**
```
./gradlew ":x-pack:qa:core-rest-tests-with-security:yamlRestTest" --tests "org.elasticsearch.xpack.security.CoreWithSecurityClientYamlTestSuiteIT.test {yaml=update/100_synthetic_source/stored text}" -Dtests.seed=78EA00D2F5CB779A -Dbuild.snapshot=false -Dtests.jvm.argline="-Dbuild.snapshot=false" -Dlicense.key=x-pack/license-tools/src/test/resources/public.key -Dtests.locale=tr-Latn-TR -Dtests.timezone=Asia/Hovd -Druntime.java=23
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.security.CoreWithSecurityClientYamlTestSuiteIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(test%20%7Byaml%3Dupdate%2F100_synthetic_source%2Fstored%20text%7D),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: Failure at [update/100_synthetic_source:126]: field [test.fields._recovery_source] should be null but was [{total=110b, total_in_bytes=110, inverted_index={total=0b, total_in_bytes=0}, stored_fields=110b, stored_fields_in_bytes=110, doc_values=0b, doc_values_in_bytes=0, points=0b, points_in_bytes=0, norms=0b, norms_in_bytes=0, term_vectors=0b, term_vectors_in_bytes=0, knn_vectors=0b, knn_vectors_in_bytes=0}] expected null, but was:<{total=110b, total_in_bytes=110, inverted_index={total=0b, total_in_bytes=0}, stored_fields=110b, stored_fields_in_bytes=110, doc_values=0b, doc_values_in_bytes=0, points=0b, points_in_bytes=0, norms=0b, norms_in_bytes=0, term_vectors=0b, term_vectors_in_bytes=0, knn_vectors=0b, knn_vectors_in_bytes=0}>
```

**Issue Reasons:**
- [main] 9 failures in test test {yaml=update/100_synthetic_source/stored text} (1.4% fail rate in 653 executions)
- [main] 5 failures in step checkpart4_release-tests (71.4% fail rate in 7 executions)
- [main] 2 failures in step part-4-entitlements (11.8% fail rate in 17 executions)
- [main] 8 failures in pipeline elasticsearch-pull-request (3.0% fail rate in 268 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -385,12 +385,6 @@ tests:
 - class: org.elasticsearch.xpack.restart.FullClusterRestartIT
   method: testWatcherWithApiKey {cluster=UPGRADED}
   issue: https://github.com/elastic/elasticsearch/issues/122061
-- class: org.elasticsearch.xpack.security.CoreWithSecurityClientYamlTestSuiteIT
-  method: test {yaml=update/100_synthetic_source/stored text}
-  issue: https://github.com/elastic/elasticsearch/issues/122100
-- class: org.elasticsearch.xpack.security.CoreWithSecurityClientYamlTestSuiteIT
-  method: test {yaml=update/100_synthetic_source/keyword}
-  issue: https://github.com/elastic/elasticsearch/issues/122101
 - class: org.elasticsearch.test.rest.ClientYamlTestSuiteIT
   method: test {yaml=snapshot.delete/10_basic/Delete a snapshot asynchronously}
   issue: https://github.com/elastic/elasticsearch/issues/122102

